src/Native/Runtime/arm64/GcProbe.S

```diff
@@ -0,0 +1,54 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include <unixasmmacros.inc>
+#include "AsmOffsets.inc"
+
+//
+// Set the Thread state and wait for a GC to complete.
+//
+// Register state on entry:
+//  x4: thread pointer
+//  
+// Register state on exit:
+//  x4: thread pointer
+//  All other registers trashed
+//
+    .global RhpWaitForGCNoAbort
+
+    .macro WaitForGCCompletion
+        ldr         w2, [x4, #OFFSETOF__Thread__m_ThreadStateFlags]
+        tst         w2, #TSF_SuppressGcStress__OR__TSF_DoNotTriggerGC
+        bne         0f
+
+        ldr         x9, [x4, #OFFSETOF__Thread__m_pHackPInvokeTunnel]
+        bl          RhpWaitForGCNoAbort
+0:
+    .endm
+
```

Comment:
```suggestion
```

Reply:
It may be best to add a new C method that does all the heavy lifting. The less assembly code we have, the better.

It would look something like:
```
NESTED_ENTRY RhpGcPollRare, _TEXT, NoHandler
    PUSH_COOP_PINVOKE_FRAME x0
    bl          RhpGcPoll2
    POP_COOP_PINVOKE_FRAME
    ret
```

```
// Standard calling convention variant and actual implementation for RhpGcPoll
void FASTCALL RhpGcPoll2(PInvokeTransitionFrame * pFrame)
{
...
}
```